torch/csrc/api/include/torch/detail/TensorDataContainer.h

```diff
@@ -136,6 +137,7 @@ AT_FORALL_SCALAR_TYPES_AND3(Bool, Half, BFloat16, TENSOR)
       sizes_({(int64_t)values.size()}), \
       scalar_type_(at::k##S), \
       type_(TensorDataContainerType::Tensor) { \
+    at::tracer::impl::NoTracerDispatchMode tracer_guard; \
```

Comment:
I'm not entirely sure whether these methods in this header file can be called in the context of tracing. @yf225 - could you please help review this part? Please check the NOTE above for the context.

Reply:
I believe the methods in this header file are strictly C++ frontend code and won't be called by JIT tracing, and it should be okay to omit this guard if we want :)